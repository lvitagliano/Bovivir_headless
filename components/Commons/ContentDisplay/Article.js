import React from 'react'
import { Container, DivImage, Link, DescContainer } from './articleStyle'
import { SubTitle, Text2 } from '../../Home/utils/commonStyles'

const Article = ({ article }) => {
  return (
    <Container flexDirection={'column'}>
      <DivImage image={article?.image.sourceUrl || `/${article.image}`} />
      <SubTitle data-testid="subtitle">{article.title}</SubTitle>
      <DescContainer style={{margin: '10px'}}>
        <Text2>{article.description}</Text2>
      </DescContainer>
      <Link data-testid="link" href={article.link.href} style={{marginBottom: '10px'}}>
        {article.link.label}
      </Link>
    </Container>
  )
}

export default Article
