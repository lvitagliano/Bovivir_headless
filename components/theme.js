import { createMuiTheme } from '@material-ui/core/styles'
import { esES } from '@material-ui/core/locale'

const primaryColor = '#762057'
const whiteColor = '#FFFFFF'
const greyColor = '#cccc'

const scroll = {
  '&::-webkit-scrollbar': {
    width: '8px',
  },
  '&::-webkit-scrollbar-thumb': {
    background: primaryColor,
    borderRadius: '8px',
    border: '1px solid #cccc',
  },
}
const { ...camposScroll } = scroll

// Create a theme instance.
const theme = createMuiTheme(
  {
    palette: {
      primary: {
        main: primaryColor,
      },
    },
    overrides: {
      MuiCssBaseline: {
        '@global': {
          body: {
            ...camposScroll,
            '& *': scroll,
          },
          // MARIDAJES
          '.containerPadreMaridajes': {
            background: '#F7F7F7',
            display: 'flex',
            justifyContent: 'space-around',
            width: '100%',
            minHeight: '100vh',
            paddingTop: '2rem',
          },
          '.maridajeContainer': {
            width: '75%',
          },
          '.filtrosMaridajeContainer': {
            minWidth: '23%',
          },
          '.cartasMaridajeGrid': {
            display: 'grid',
            gridTemplateColumns: 'repeat(auto-fit, 18rem)',
            justifyContent: 'center',
            alignContent: 'space-evenly',
            gap: '0.3rem',
            padding: '0.5rem 0',
          },
          '.hoverCartasMaridaje': {
            position: 'absolute',
            top: '0',
            left: '0',
            padding: '15px 30px 20px',
            width: '100%',
            height: '100%',
            background: 'rgba(118,32,87,0.7)',
            color: '#fff',
            opacity: '0',
            transition: 'all 700ms ease-out',

            '&:hover': {
              opacity: '1',
              borderRadius: '1%',
            },
          },
          // MARIDAJES
        },
      },
      MuiCheckbox: {
        root: {
          color: primaryColor,
          padding: '0px',
        },
        colorSecondary: {
          '&$checked': {
            color: primaryColor,
          },
        },
      },
    },
  },
  esES
)

export default theme
