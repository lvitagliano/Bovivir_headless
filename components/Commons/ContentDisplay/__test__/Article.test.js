import React from 'react';
import ReactDOM from 'react-dom';
import Article from "../Article";
import {render,cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";
import '@testing-library/jest-dom';
import { ThemeProvider } from 'styled-components';
import theme from "../../../../Styles/themes/main";

afterEach(cleanup);

const article = {
    title: 'Titulo item 1',
    label:'Bonvivir',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image: 'https://images.pexels.com/photos/3019019/pexels-photo-3019019.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260',
    link:{label:'Link Test',href:'/test_link'}
};

afterEach(cleanup);

it("renders without problems", () => {
  const div = document.createElement("div");
  ReactDOM.render(
  <ThemeProvider theme={theme}>
    <Article article={article}/>
 </ThemeProvider>
 ,div);
});

it("renders correctly", () => {
  const domElem = render(
    <ThemeProvider theme={theme}>
       <Article article={article}/>
    </ThemeProvider>
    );
  expect(domElem.getByTestId("subtitle")).toBeInTheDocument();
  domElem.getByText("Lorem ipsum dolor sit amet, consectetur adipiscing elit.")
  expect(domElem.getByTestId("link")).toHaveAttribute('href',"/test_link");
});

it("matches snapshot Article", () => {
  const elem = renderer.create(
  <ThemeProvider theme={theme}>
    <Article article={article}/>
 </ThemeProvider>
 );
  expect(elem).toMatchSnapshot();
});

