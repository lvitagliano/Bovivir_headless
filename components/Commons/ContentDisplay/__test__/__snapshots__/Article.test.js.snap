// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot Article 1`] = `
<div
  className="articleStyle__Container-tnp56t-0 dbTQlN"
>
  <img
    className="articleStyle__DivImage-tnp56t-1 iUQLGN"
    src="https://images.pexels.com/photos/3019019/pexels-photo-3019019.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260"
  />
  <p
    className="commonStyles__SubTitle-g2rlbn-7 eSUazo"
    data-testid="subtitle"
  >
    Titulo item 1
  </p>
  <div
    className="articleStyle__DescContainer-tnp56t-3 eKbERj"
  >
    <p
      className="commonStyles__Text2-g2rlbn-2 bRmgTK"
      color="#888"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
  </div>
  <a
    className="articleStyle__Link-tnp56t-2 xVjgb"
    data-testid="link"
    href="/test_link"
  >
    Link Test
  </a>
</div>
`;
